src/docfx/lib/git/CommitBuildTimeProvider.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.Docs.Build
+{
+    internal class CommitBuildTimeProvider
+    {
+        private readonly Repository _repo;
+        private readonly string _commitBuildTimePath;
+        private readonly IReadOnlyDictionary<string, DateTime> _buildTimeByCommit;
+
+        public CommitBuildTimeProvider(Repository repo)
```

Comment:
considering the submodule case?

Reply:
added a TODO

Reply:
what's the meaning of current Docs.Build implementation?